bindings/python/iota_sdk/types/block/wrapper.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2023 IOTA Stiftung
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+from dataclasses import dataclass
+from typing import Union
+from iota_sdk.types.block.basic import BasicBlock
+from iota_sdk.types.block.validation import ValidationBlock
+from iota_sdk.types.common import HexStr, json
+from iota_sdk.types.signature import Ed25519Signature
+from iota_sdk.utils import Utils
+
+
+@json
+@dataclass
+class BlockWrapper:
+    """A block wrapper type that can hold either a `BasicBlock` or a `ValidationBlock`.
+    Shared data is stored alongside such a block in the `BlockHeader` and `Signature` fields.
+
+    Attributes:
+        header: The Block header.
+        block: Holds either a `BasicBlock` or a `ValidationBlock`.
+        signature: The Block signature.
+    """
+    protocol_version: int
+    network_id: str
+    issuing_time: str
+    slot_commitment_id: HexStr
+    latest_finalized_slot: int
+    issuer_id: HexStr
+    block: Union[BasicBlock, ValidationBlock]
+    signature: Ed25519Signature
```

Comment:
Do we not have a generic Signature type?

Reply:
yeah, we do, and also we have an abstract `Block` type, but then we have to manually deserialize and the code becomes really ugly.

Reply:
Not sure to follow but if we have a Signature type, we should use it. Ed25519Signature s not really correct.

Reply:
if there's going to be another signature type we can do the same as for the block. We would then change line 32 to:
```python
signature: Union[Ed25519Signature, WinternitzOneTimeSignatureForTheLolz]
```
and can keep using the auto-implemented deserialization logic.

Reply:
I am not Pythonean enough, but I would say it's possible to use the abstract types (i.e. `Block`, `Signature`) and then manually implement `from_dict` where we read the `type` and then deserialize accordingly, but as I said ... that's a lot of custom logic that's not needed atm. We can always do this when the time is ripe and right.

Reply:
lets change it to the `signature: Union[Ed25519Signature]`, which is also in line with other cases like when we have Outputs